segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/ServerConnectionInboundHandler.java

```diff
@@ -56,9 +55,20 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
 
     @Override
     public void send(WireCommand cmd) {
-        getChannel().writeAndFlush(cmd).addListener(ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE);
+        Channel c = getChannel();
+        // Work around for https://github.com/netty/netty/issues/3246
+        EventLoop eventLoop = c.eventLoop();
+        if (eventLoop.inEventLoop()) {
+            eventLoop.execute(() -> writeAndFlush(c, cmd));
```

Comment:
How does this guarantee happens-before? Is this a guarantee of every `EventLoop`? (would be nice to see in the `EventLoop` javadoc if the case).